### PR TITLE
[Enhancement] support lake compaction validation

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kevincai — DISMISSED** [review]

**Review by sduzh — DISMISSED** [review]

**Review by sduzh — DISMISSED** [review]

**Review by tracymacding — APPROVED** [review]

**Review by sduzh — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

File: be/src/common/config.h [hunk 1]
```diff
@@ -953,6 +953,8 @@ CONF_mInt32(starlet_delete_files_max_key_in_batch, "1000");
 CONF_mInt64(lake_metadata_cache_limit, /*2GB=*/"2147483648");
 CONF_mBool(lake_print_delete_log, "false");
 CONF_mInt64(lake_compaction_stream_buffer_size_bytes, "1048576"); // 1MB
+// The interval to check whether lake compaction is valid. Set to <= 0 to disable the check.
+CONF_mInt32(lake_compaction_check_valid_interval_minutes, "30"); // 30 minutes
```
Comment (trueeyu) [inline be/src/common/config.h:957]:
```
eyu@eyu:~/code/sr2$ grep "_min," be/src/common/config.h
CONF_Int32(streaming_load_thread_pool_num_min, "0");
CONF_Int32(fragment_pool_thread_num_min, "64");
CONF_mInt32(scan_context_gc_interval_min, "5");
eyu@eyu:~/code/sr2$ grep "_minute," be/src/common/config.h
eyu@eyu:~/code/sr2$ grep "_minutes," be/src/common/config.h
```

Comment (trueeyu) [inline be/src/common/config.h:957]: why not use seconds?

**Review by kevincai — APPROVED** [review]

File: be/src/common/config.h (hunk 1 above)
Comment (starrocks-xupeng) [reply be/src/common/config.h:957]: compaction sometimes is very slow in shared-data mode, maybe take 1 hour if reads data from s3, so use minute

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java [hunk 2]
```diff
@@ -452,6 +452,22 @@ public void cancelCompaction(long txnId) {
         }
     }
 
+    protected ConcurrentHashMap<PartitionIdentifier, CompactionJob> getRunningCompactions() {
+        return runningCompactions;
+    }
+
+    public boolean existCompaction(long txnId) {
+        for (Iterator<Map.Entry<PartitionIdentifier, CompactionJob>> iterator = getRunningCompactions().entrySet().iterator();
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java:460]: Why does map need to be completely iterated?

Comment (starrocks-xupeng) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java:460]: txnId is not map key

**Review by trueeyu — APPROVED** [review]

**Comment by starrocks-xupeng** [comment]
@mergify backport branch-3.2

**Pull request merged**